impl/src/main/java/org/jboss/cdi/tck/tests/full/extensions/lifecycle/atd/AfterTypeDiscoveryTest.java

```diff
@@ -78,16 +81,35 @@ public void testInitialInterceptors() {
     @Test
     @SpecAssertions({ @SpecAssertion(section = AFTER_TYPE_DISCOVERY, id = "b"), @SpecAssertion(section = AFTER_TYPE_DISCOVERY, id = "ha") })
     public void testInitialAlternatives() {
-        assertEquals(extension.getAlternatives().size(), 3);
-        assertEquals(extension.getAlternatives().get(0), AlphaAlternative.class);
-        assertEquals(extension.getAlternatives().get(1), EchoAlternative.class);
-        assertEquals(extension.getAlternatives().get(2), DeltaAlternative.class);
+        // frameworks might add their own alternatives, we cannot assert positions in list but rather just ordering
+        assertTrue(extension.getAlternatives().size() >= 3);
+        List<Class<?>> alternatives = extension.getAlternatives();
+        Integer alphaAltIndex = null;
+        Integer echoAltIndex = null;
+        Integer deltaAltIndex = null;
+        for (int i = 0; i < alternatives.size(); i++) {
+            if (alternatives.get(i).equals(AlphaAlternative.class)) {
+                alphaAltIndex = i;
+            }
+            if (alternatives.get(i).equals(EchoAlternative.class)) {
+                echoAltIndex = i;
+            }
+            if (alternatives.get(i).equals(DeltaAlternative.class)) {
+                deltaAltIndex = i;
+            }
+        }
+        assertNotNull(alphaAltIndex);
+        assertNotNull(echoAltIndex);
+        assertNotNull(deltaAltIndex);
+        assertTrue(alphaAltIndex < echoAltIndex);
+        assertTrue(echoAltIndex < deltaAltIndex);
     }
 
     @Test
     @SpecAssertions({ @SpecAssertion(section = AFTER_TYPE_DISCOVERY, id = "d"), @SpecAssertion(section = AFTER_TYPE_DISCOVERY, id = "hc") })
     public void testInitialDecorators() {
-        assertEquals(extension.getDecorators().size(), 4);
+        assertTrue(extension.getDecorators().size() >= 4);
+        // TODO same issue?
```

Comment:
I'll perform similar change here for decorators as soon as we have confirmation that this was indeed the issue.

Reply:
Looking at `testInitialInterceptors` and it just asserts presence, do we really have to assert order for alternatives and decorators? (And if so, why not for interceptors? :-) )

Reply:
I think we check the ordering of interceptors in separate test (or rather, interceptors TCKs do that) but I haven't tried to pinpoint that, I just vaguely recall seeing such test.
It makes sense to assert ordering of alternatives/decorators as that is being manipulated in the extension. For instance decorator ordering is being reversed as part of the test.

I have altered the decorator logic in a similar way, so this PR should now be ready for review.
Also did a quick check with embedded Weld and it looks fine.

Reply:
OK, thanks. I was just wondering :-)